builtin/provisioners/chef-client/ssh_provisioner.go

```diff
@@ -0,0 +1,72 @@
+package chefclient
+
+import (
+	"bytes"
+	"strings"
+
+	"github.com/hashicorp/terraform/communicator"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func (p *Provisioner) sshInstallChefClient(
+	o terraform.UIOutput,
+	comm communicator.Communicator) error {
+	var installCmd bytes.Buffer
+
+	// Build up a single command based on the given config options
+	installCmd.WriteString("curl")
+	if p.HTTPProxy != "" {
+		installCmd.WriteString(" --proxy " + p.HTTPProxy)
+	}
+	if p.NOProxy != nil {
+		installCmd.WriteString(" --noproxy " + strings.Join(p.NOProxy, ","))
+	}
+	installCmd.WriteString(" -LO https://www.chef.io/chef/install.sh 2>/dev/null &&")
```

Comment:
Yeah, fair enough... Will remove the `2>/dev/null` part. I added it because the console is becoming quite chatty but for debugging I can see this might become annoying at a certain point... Will also make the URL a const.